src/engine/GroupBy.cpp

```diff
@@ -933,24 +989,8 @@ void GroupBy::createResultFromHashMap(
             sparqlExpression::copyExpressionResult(expressionResult);
 
         // Extract values
-        auto visitor =
-            [&evaluationContext, &result, &i, &localVocab, &alias,
-             &currentBlockSize]<sparqlExpression::SingleExpressionResult T>(
-                T&& singleResult) mutable {
-              auto generator = sparqlExpression::detail::makeGenerator(
-                  std::forward<T>(singleResult), currentBlockSize,
-                  &evaluationContext);
-
-              // TODO<C++23> use views::enumerate
-              auto idx = 0;
-              for (sparqlExpression::IdOrString val : generator) {
-                (*result)(i + idx++, alias.outCol_) =
-                    sparqlExpression::detail::constantExpressionResultToId(
-                        std::move(val), *localVocab);
-              }
-            };
-
-        std::visit(visitor, std::move(expressionResult));
+        extractValues(std::move(expressionResult), evaluationContext, result,
+                      localVocab, alias.outCol_);
```

Comment:
We have some unnecessary copies here of the `expressionResult` but that is something for another PR as we have to do some refactoring in the expression module, because have to know, whether we still need the contained `string` values...